normandy/recipes/tests/test_models.py

```diff
@@ -740,6 +740,40 @@ def test_it_rollbacks_changes_if_error_happens_on_unpublish(self, mocked_remotes
         saved = Recipe.objects.get(id=recipe.id)
         assert saved.approved_revision.enabled
 
+    def test_enable_rollback_enable_rollout_invariance(self):
+        rollout_recipe = RecipeFactory(
+            name="Rollout",
+            approver=UserFactory(),
+            enabler=UserFactory(),
+            action=ActionFactory(name="preference-rollout"),
+            arguments={"slug": "myslug"},
+        )
+        assert rollout_recipe.enabled
+
+        rollback_recipe = RecipeFactory(
+            name="Rollback",
+            action=ActionFactory(name="preference-rollback"),
+            arguments={"rolloutSlug": "myslug"},
+        )
+        approval_request = rollback_recipe.latest_revision.request_approval(creator=UserFactory())
+        approval_request.approve(approver=UserFactory(), comment="r+")
```

Comment:
Isn't it possible to do these two lines with `RecipeFactory(..., approver=UserFactory())`?

Reply:
I honestly don't understand what you mean. You can't create a new recipe but you have to create a new approval request. No?

Reply:
Hm. I'm not sure if I'm confused or not. What I meant was this:

```diff
 rollback_recipe = RecipeFactory(
      name="Rollback",
      action=ActionFactory(name="preference-rollback"),
      arguments={"rolloutSlug": "myslug"},
+     approver=UserFactory(),
 )
-approval_request = rollback_recipe.latest_revision.request_approval(creator=UserFactory())
-approval_request.approve(approver=UserFactory(), comment="r+")
```

Does that work? Or am I missing something?

Reply:
Ah! I see. I went down there too. 

You can't create a rollback recipe unless you have a rollout recipe. And you can't create a new rollback recipe (to match a rollout recipe) if there's already another rollback recipe by that `rolloutSlug`. 

Hence, I "revive" that initial `rollback_recipe` instance.

Reply:
I really don't follow anything you just said. I understand that you can't create a rollback without a rollout, but there is already a rollout. I also understand that you can't have a duplicate rollback recipe, but that's ok too because you are only making a single rollout recipe.

I'm trying to figure out why when you make the first recipe, you create and approve it in one step, but then when you make the second recipe, you create and approve it in two steps. I don't think this has anything to do with the intricacies of the particular actions. It just seems like you are being inconsistent.

Feel free to merge this as is though.

Reply:
I also don't understand what you're reviving. There is only one rollback being created here, and it's all being done at once, just in three function calls instead of one.